gettid() is undeclared (Linux)

### Describe the bug

A test fails to compile due to the `gettid()` function being undeclared.

### Reproduction Steps

`v vlib/v/tests/trace_calls_test.v`

### Expected Behavior

The test to succeed.

### Current Behavior

The test fails to compile:

```
 FAIL  [2077/2214] C:   447.2 ms, R:   750.467 ms vlib/v/tests/trace_calls_test.v
> trace_fns_only_println compilation output:
================== C compilation error (from cc): ==============
cc: /tmp/v_1001/tsession_7f5da030e740_01JGMW261EACT7W3A17Q67E947/single_println.01JGMW7YBVE9N13HGQ7BHP4SR6.tmp.c:14022:10: error: call to undeclared function 'gettid'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
cc:  14022 |                         tid = gettid();
cc:        |                               ^
cc: 1 error generated.
================================================================
```

### Possible Solution

https://stackoverflow.com/questions/61979901/gcc-warns-about-gettid-syscall-wrapper-with-glibc-2-30-8

### Additional Information/Context

_No response_

### V version

0.4.9

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.9 3953445
OS: linux, "OpenMandriva Lx 24.90" (VM)
Processor: 8 cpus, 64bit, little endian, AMD Ryzen 9 5900X 12-Core Processor

getwd: /tmp/v-0.4.9
vexe: /tmp/v-0.4.9/v
vexe mtime: 2025-01-03 01:18:56

vroot: OK, value: /tmp/v-0.4.9
VMODULES: OK, value: /home/user/.vmodules
VTMP: OK, value: /tmp/v_1001

Git version: git version 2.47.1
Git vroot status: 08d5c282-dirty
.git/config present: false

CC version: OpenMandriva 19.1.6-1 clang version 19.1.6 (/builddir/build/BUILD/llvm-19.1.6-build/llvm-project-llvmorg-19.1.6/clang 1cc1c8b2641ccba94c7b90b5e4c4e500729f7556)
emcc version: N/A
thirdparty/tcc: N/A
```

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

<sub><a href="https://huly.app/guest/vlang-66f40c4d-a476b54c67-771fdd?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2Nzc3NDcwYWYxNDRmNTg1YWU1MjQxNmMiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctYWxleGFuZGVyLXZsYW5nLTY2ZjQwYzRkLWE0NzZiNTRjNjctNzcxZmRjIn0.Mf7Uuov6fHBc7WeTIQrPWbayz-MyLJOWfX16UWYFFJk">Huly&reg;: <b>V_0.6-21782</b></a></sub>

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -906,6 +906,10 @@ pub fn (mut g Gen) init() {
 #endif
 ')
 		}
+		if g.pref.os == .linux {
+			// For gettid() declaration (and other GNU-specific bits).
+			g.cheaders.writeln('#define _GNU_SOURCE')
+		}
 		if g.pref.os == .wasm32 {
 			g.cheaders.writeln('#define VWASM 1')
 			// Include <stdint.h> instead of <inttypes.h> for WASM target

